islandora-indexing-fcrepo/src/main/java/ca/islandora/alpaca/indexing/fcrepo/FcrepoIndexer.java

```diff
@@ -130,10 +151,81 @@ public void configure() {
                                 "Received 404 from Milliner, skipping de-indexing."
                         )
                         .end()
-                .setProperty("urn").jsonpath("$.object.id")
-                .setProperty("uuid").simple("${exchangeProperty.urn.replaceAll(\"urn:uuid:\",\"\")}")
+                // Parse the event into a POJO.
+                .unmarshal().json(JsonLibrary.Jackson, AS2Event.class)
+
+                // Extract relevant data from the event.
+                .setProperty("event").simple("${body}")
+                .setProperty("uuid").simple("${exchangeProperty.event.object.id.replaceAll(\"urn:uuid:\",\"\")}")
+
+                // Prepare the message.
                 .removeHeaders("*", "Authorization")
                 .setHeader(Exchange.HTTP_METHOD, constant("DELETE"))
-                .toD(getMillinerBaseUrl() + "resource/${exchangeProperty.uuid}");
+                .transform(simple("${null}"))
+
+                // Remove the file from Gemini.
+                .toD(getMillinerBaseUrl() + "node/${exchangeProperty.uuid}");
+
+        from("{{media.stream}}")
+                .routeId("FcrepoIndexerMedia")
+
+                // Parse the event into a POJO.
+                .unmarshal().json(JsonLibrary.Jackson, AS2Event.class)
+
+                // Extract relevant data from the event.
+                .setProperty("event").simple("${body}")
+                .setProperty("sourceField").simple("${exchangeProperty.event.attachment.content.sourceField}")
+                .setProperty("jsonUrl").simple("${exchangeProperty.event.object.url[1].href}")
+
+                // Prepare the message.
+                .removeHeaders("*", "Authorization")
+                .setHeader(Exchange.HTTP_METHOD, constant("POST"))
+                .setHeader("Content-Location", simple("${exchangeProperty.jsonUrl}"))
+                .transform(simple("${null}"))
```

Comment:
Could this just be `.setBody(simple("${null}"))` instead of a transform?

Reply:
I'm not 100% sure what the difference is, but "setBody" seems clearer to me.  I'll have to try it out to know for sure.

Reply:
I think `transform()` is part of the [message translator EIP](https://github.com/apache/camel/blob/master/camel-core/src/main/docs/eips/message-translator.adoc), so if you are just wanting to make the body of the message empty I think `.setBody()` would be more efficient.

But I don't know for sure

Reply:
Whelp, I did it and tested it and nothing seems to have changed, so I'm pushing it up.  It just reads better IMO.